### PR TITLE
feat(ui): adding status color mapping for k8s

### PR DESCRIPTION
**Pull request by BillyFigueroa**
## Description
Adding color mapping for the various resources for k8s

## Related Issue

- #129 

## Screenshots
Screenshot 2024-08-29 at 1 32 23 PM

### CODE REVIEW

File: ui/src/lib/features/k8s/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  },
+}
+
+// Function to get the color and status for a specific type and status
+export const getColorAndStatus = <T extends keyof K8StatusMapping>(
+  type: T,
+  status: keyof K8StatusMapping[T],
+): string => {
+  return ((k8StatusMapping[type][status] as { color: string }).color as string) || 'Unknown'
+}
```
Comment (TristanHoladay) [inline ui/src/lib/features/k8s/helpers.ts:86]:
```suggestion
export const getColorAndStatus = <T extends keyof K8StatusMapping>(
  type: T,
  status: keyof K8StatusMapping[T],
) => {
  return ((k8StatusMapping[type][status] as { color: string }).color) || 'Unknown'
}
```

File: ui/src/lib/features/k8s/workloads/pods/status/component.test.ts [hunk 2]
```diff
@@ -6,11 +6,11 @@ describe('Status component', () => {
...
     expect(container.firstChild).toHaveClass('text-green-400')
   })
-  test('renders text-red-400 for Running', () => {
-    const { container } = render(Status, { props: { status: 'Stopped' } })
+  test('renders text-red-400 for Failed', () => {
+    const { container } = render(Status, { props: { status: 'Failed' } })
     expect(container.firstChild).toHaveClass('text-red-400')
   })
-  test('renders text-yellow-500 for Running', () => {
+  test('renders text-yellow-500 for Pending', () => {
```
Comment (TristanHoladay) [inline ui/src/lib/features/k8s/workloads/pods/status/component.test.ts:13]:
```suggestion
  test('renders text-orange-300 for Pending', () => {
```

File: ui/src/lib/features/k8s/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+export type StatusColorMapping = {
+  success: string
+  info: string
+  warning: string
+  error: string
+  disabled: string
+}
+
+const statusColors: StatusColorMapping = {
```
Comment (TristanHoladay) [inline ui/src/lib/features/k8s/helpers.ts:11]: what do you think about getting rid of the `StatusColorMapping` type and instead doing `as const` on `statusColors` so we can see the class string in intellisense?

Screenshot from 2024-08-30 08-40-13

Screenshot from 2024-08-30 08-38-11

File: ui/src/lib/features/k8s/helpers.ts (hunk 1 above)
Comment (BillyFigueroa) [reply ui/src/lib/features/k8s/helpers.ts:86]: Nice little cleanup. I ll make the change

File: ui/src/lib/features/k8s/helpers.ts (hunk 3 above)
Comment (BillyFigueroa) [reply ui/src/lib/features/k8s/helpers.ts:11]: I just watched a video this morning on how you can use `as const` in this way to sort of replicate enum's as a way to not use enums

I ll make the change

Comment (TristanHoladay) [reply ui/src/lib/features/k8s/helpers.ts]: nice! i was trying to remember that exact thing about enums lol. i don't think i've had the opportunity before to use `as const` but i like it.

Comment (BillyFigueroa) [reply ui/src/lib/features/k8s/helpers.ts]: > nice! i was trying to remember that exact thing about enums lol. i don't think i've had the opportunity before to use `as const` but i like it.

very neat. I dig it

**Review by TristanHoladay — APPROVED** [review]

**Pull request merged**